### DISCUSSION TITLE
Page using getStaticProps does not seem to be fetching fresh data with `swr`

### DISCUSSION THREAD

**Discussion by brianlovin** [vercel/next.js#11286] [Help] [answered]
Hey team, I'm hitting some issues with `swr` not running on the initial mount of a Next.js application running with `getStaticProps` for SSG. 

Here's a video of this in action: https://www.dropbox.com/s/bchsi0qx1pcate6/swr-gsp.mp4?dl=0

What I'm doing:
- load https://brianlovin.com/bookmarks – initial render is old data that was computed at build time
- see the (expired) first link
- unfocus / refocus the tab and you can see `swr` kick off and get fresh data from the database

My question is why is `swr` not getting fresh data on the initial mount, and only on refocus? My impression is that it should run for _both_ events. 

URL where it's happening: https://brianlovin.com/bookmarks
Code: 

- page: https://github.com/brianlovin/brian-lovin-next/blob/master/src/pages/bookmarks.tsx
- fetcher: https://github.com/brianlovin/brian-lovin-next/blob/master/src/api/index.ts

Edit: it seems entirely more simple to just migrate this specific page to use `getServerSideProps` – in fact, I'll probably make this change right now. But the question still stands, especially as static generation becomes more common in Next.js land, so I'll leave this post up for possible tips on making these two tools work better together!

**Comment by pacocoursey** [comment, marked as answer]
At first glance it looks like SWR doesn't do a client side revalidation on initial mount if you already specified `initialData`. cc @quietshu as maybe we should update/clarify this behavior.

https://github.com/zeit/swr/blob/8991780b564082cd08a103073e9584cbb3ce6eb7/src/use-swr.ts#L388

Instead, you could `mutate(BOOKMARKS)` (manual revalidation) inside a mount `useEffect` for that page.

**Reply by brianlovin** [reply]
This is the fix, tested and confirmed. It seems like this could potentially be a common issue more people run into, maybe it's worth providing a flag to override that behavior? Either way, I'm going to update the rest of my pages where initialData is included + SSG in order to always show fresh data. Thank you!

**Reply by lkbr** [reply]
For people who use initialData with SSR, the current default is preferred (at least for me). I don't want to "double" fetch data on the client that I just fetched on the server. I can manually revalidate  if need to e.g. the SSR data is from a cache that might be a bit old.

**Reply by brianlovin** [reply]
> I can manually revalidate if need to e.g. the SSR data is from a cache that might be a bit old.

Great point. The double fetching is potentially wasteful, in the case that the SSR data is _actually_ the freshest. For my personal case with SSG, however, this puts too much pressure on thinking about _when has the latest deployment gone out_ versus _when has the latest content been updated_. 

Anyways, good to have a solid escape hatch for now :)

**Reply by shuding** [reply]
Yeah like what @lkbr said, `initialData` is mostly used together with SSR or local state management. In those cases it makes sense to skip the initial fetch. 

You can also use the bounded `mutate` function returned by `useSWR`:

```js
const { data, error, mutate } = useSWR(BOOKMARKS, query => fetcher({ query }), { initialData: props.data })

React.useEffect(() => mutate(), [])
```

**Comment by raheesm** [comment]
add the useSwr() inside useEffect
